src/core/core.controller.js

```diff
@@ -469,9 +469,14 @@ class Chart {
 		// Make sure dataset controllers are updated and new controllers are reset
 		const newControllers = me.buildOrUpdateControllers();
 
+		me.notifyPlugins('beforeElementUpdate');
```

Comment:
Not sure what it would be used for, but I have no strong objections

Reply:
Shouldn't it be `beforeElementsUpdate`? 
we have `beforeDatasetsUpdate` and `beforeDatasetUpdate`, so in that context one could think this would fire for each element if its not plural.

Reply:
Good point, I've renamed it to `beforeElementsUpdate`.